library/Icinga/Util/Csp.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Icinga\Util;
+
+use Icinga\Web\Response;
+use Icinga\Web\Session;
+use Icinga\Web\Window;
+
+class Csp
+{
+    /** @var self */
+    private static $instance = null;
+
+    /** @var string */
+    private $styleNonce = null;
+
+    /** @var string */
+    private $scriptNonce = null;
+
+    private function loadNoncesFromSession()
+    {
+        $instance = self::getInstance();
+        $windowSession = Window::getInstance()->getSessionNamespace('csp');
+
+        $sessionStyleNonce = $windowSession->get('csp_style_nonce');
+        $sessionScriptNonce = $windowSession->get('csp_script_nonce');
+
+        if (! empty($sessionStyleNonce)) {
+            $instance->styleNonce = $sessionStyleNonce;
+        }
+
+        if (! empty($sessionScriptNonce)) {
+            $instance->scriptNonce = $sessionScriptNonce;
+        }
+    }
+
+    private static function getInstance(): self
+    {
+        if (self::$instance === null) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    public static function getStyleNonce(): ?string
```

Comment:
This must throw an exception. If I request a nonce, there has to be a nonce. Else we have bugs in our code.